server/Controllers/changePassword.js

```diff
@@ -0,0 +1,45 @@
+import bcrypt from "bcrypt";
+import UserModel from "../Schemas/User";
+
+const validatePassword = (password) => {
+  const reg = /(?=^.{8,}$)((?!.*\s)(?=.*[A-Z])(?=.*[a-z]))((?=(.*\d){1,})|(?=(.*\W){1,}))^.*$/;
+  return reg.test(password);
+};
+
+const changePassword = async (req, res) => {
+  const encodedNewPassword = bcrypt.hashSync(req.body.newPassword, 10);
+
+  if (!(await validatePassword(req.body.newPassword))) {
+    res.status(400).send();
+  } else {
+    try {
+      const userInfos = await UserModel.findById(req.userId);
+      if (userInfos === null) {
+        res.status(404).send();
+      } else if (
+        !(await bcrypt.compareSync(req.body.oldPassword, userInfos.password))
```

Comment:
Use the async version of compare !! (and using await with the Sync version doesn't make sense)

I thought we already talk about that 😕... from the bcrypt doc:
![image](https://user-images.githubusercontent.com/24413792/78744286-7af6ba80-7961-11ea-991d-a40c7de1bdd2.png)